pyccel/parser/base.py

```diff
@@ -500,17 +512,112 @@ def insert_import(self, expr):
                         container[source] = []
                     container[source] += name
 
-    def print_namespace(self):
+    def dump(self, filename=None):
+        """
+        Dump the current ast using Pickle.
+
+          Parameters
+          ----------
+          filename: str
+            output file name. if not given `name.pyccel` will be used and placed
+            in the Pyccel directory ($HOME/.pyccel)
+        """
+        import pickle
+        import hashlib
+
+        if not filename:
+            if not self.filename:
+                raise ValueError('Expecting a filename to load the ast')
+
+            path , name  = os.path.split(self.filename)
+
+            if not name.split('.')[-1] == 'pyh':
+                return
+            else:
+                name = name[:-4]
+
+            name     = '{}.pyccel'.format(name)
+            filename = os.path.join(path, name)
+        # check extension
+
+        if not filename.split(""".""")[-1] == 'pyccel':
+            raise ValueError('Expecting a .pyccel extension')
+
+#        print('>>> home = ', os.environ['HOME'])
+        # ...
+
+        # we are only exporting the AST.
+        try:
+            code = self.code.encode('utf-8')
+            hs   = hashlib.md5(code)
+            with open(filename, 'wb') as f:
+                pickle.dump((hs.hexdigest(), __version__, self), f, pickle.HIGHEST_PROTOCOL)
+        except FileNotFoundError:
+            pass
+        except PermissionError:
+            pass
+
+    def load(self, filename=None):
+        """ Load the current ast using Pickle.
+
+          Parameters
+          ----------
+          filename: str
+            output file name. if not given `name.pyccel` will be used and placed
+            in the Pyccel directory ($HOME/.pyccel)
+        """
+
+        # ...
+        import pickle
+
+        if not filename:
+            if not self.filename:
+                raise ValueError('Expecting a filename to load the ast')
+
+            path , name = os.path.split(self.filename)
 
-        # TODO improve spacing
+            if not name.split('.')[-1] == 'pyh':
+                return
+            else:
+                name = name[:-4]
+
+            name     = '{}.pyccel'.format(name)
+            filename = os.path.join(path, name)
+
+        if not filename.split(""".""")[-1] == 'pyccel':
+            raise ValueError('Expecting a .pyccel extension')
 
-        print ('------- namespace -------')
-        for (k, v) in self.namespace.items():
-            print ('{var} \t :: \t {dtype}'.format(var=k, dtype=type(v)))
-        print ('-------------------------')
+        try:
+            with open(filename, 'rb') as f:
+                hs, version, parser = pickle.load(f)
+        except Exception:
```

Comment:
I agree with codacy (once I switched back to our pylint config file). We should never catch `Exception` unless we are raising an error. Maybe you want `OSError` here?

Reply:
the exception depends on pickle , it can be what ever exception pickle raises

Reply:
In that case shouldn't the try/except just be around the pickle line?
https://pythontic.com/modules/pickle/exceptions#:~:text=Exceptions%20provided%20by%20the%20Python%20Pickle%20Module%3A&text=PicklingError%20is%20used%20when%20the,is%20not%20able%20to%20proceed.&text=This%20is%20exception%20pickle.,object%20hierarchie(s)%20fails.
It looks like it should be `except pickle.PickleError`

Reply:
so we either keep it, or delete the try except clause

Reply:
it can also be the fact that the file doesn't exist or that we don't have the permission

Reply:
I added the pickle error , so it should be fine